src/test/kotlin/com/cjbooms/fabrikt/generators/MicronautControllerGeneratorTest.kt

```diff
@@ -67,6 +66,21 @@ class MicronautControllerGeneratorTest {
         assertThat(controllers).isEqualTo(expectedControllers)
     }
 
+    @Test
+    fun `correct models are generated for ControllerCodeGenOptionType_AUTHENTICATION`() {
+        val basePackage = "examples.authentication"
+        val api = SourceApi(readTextResource("/examples/authentication/api.yaml"))
+        val expectedControllers = readTextResource("/examples/authentication/controllers/micronaut/Controllers.kt")
+
+        val controllers = MicronautControllerInterfaceGenerator(
+            Packages(basePackage),
+            api,
+            setOf(ControllerCodeGenOptionType.AUTHENTICATION)
+        ).generate().toSingleFile()
+
+        assertThat(controllers).isEqualTo(expectedControllers)
+    }
+
```

Comment:
**praise:** Good :)

If you want to be more fancy and learn something new, this could be done really well with a `TestFactory`. But just a side note; imo it's fine as it is now.